File: backend/src/models/Post.js
const mongoose = require('mongoose');

const postSchema = new mongoose.Schema(
  {
    author: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true, index: true },
    content: { type: String, maxlength: 2200, default: '' },
    images: [{ type: String }],
    likes: [{ type: mongoose.Schema.Types.ObjectId, ref: 'User' }],
    likesCount: { type: Number, default: 0 },
    commentsCount: { type: Number, default: 0 },
    isActive: { type: Boolean, default: true },
  },
  { timestamps: true }
);

postSchema.index({ createdAt: -1 });

module.exports = mongoose.model('Post', postSchema);

File: app/tmzon-android/app/src/main/java/app/tmzon/model/Post.java
package app.tmzon.model;

import com.google.gson.annotations.SerializedName;
import java.util.List;

public class Post {
    @SerializedName("_id") public String id;
    public String content;
    public User author;
    public List<String> likes;
    public List<Comment> comments;
    public String createdAt;
}
